portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java

```diff
@@ -23,43 +29,47 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.List;
 
-@WebServlet("/comments")
-public final class BlogCommentsServlet extends HttpServlet {
-
-    // Comments data
-    ArrayList<BlogComment> commentsList = new ArrayList<BlogComment>();
+// Servlet to load and list all previous comments
+@WebServlet("/load-comments")
+public final class LoadCommentsServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    // Convert the comments data to JSON
-    String json = convertToJsonUsingGson(commentsList);
-
-    // Send the JSON as the response
-    response.setContentType("application/json");
-    response.getWriter().println(json);
-  }
+    // Get comments sorting order 
+    String sortingOrder = request.getParameter("sort");
 
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if(sortingOrder == null) {
+        sortingOrder = "SortDirection.DESCENDING";
+    }
 
-    // Get the input from the form.
-    Date date = new Date();
-    String text = request.getParameter("comment-text");
+    // Get BlogComment entities from Datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    Query query = new Query("BlogComment").addSort("postDate", sortingOrder);
+    PreparedQuery results = datastore.prepare(query);
 
-    // Upload comment 
-    uploadComment(date, text);
+    // Create ArrayList<BlogComment> from entities
```

Comment:
Seems the comment is not needed here as it is obvious from code.

Reply:
Got you 😄